hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/service/TestSnapshotDeletingService.java

```diff
@@ -0,0 +1,301 @@
+        OmBucketInfo.newBuilder().setVolumeName(volumeName)
+            .setBucketName(bucketName)
+            .setBucketLayout(bucketLayout)
+            .build());
+
+    return createKey(volumeName, bucketName, keyName);
+  }
+
+
+  private int createSnapshotDataForBucket1() throws Exception {
+    int snapshotCount = 0;
+    OmKeyArgs bucket1key1 = createVolumeBucketKey(VOLUME_NAME, BUCKET_NAME_ONE,
+        BucketLayout.DEFAULT, "bucket1key1");
+
+    createSnapshot(VOLUME_NAME, BUCKET_NAME_ONE, "bucket1snap1",
+        ++snapshotCount);
+
+    OmKeyArgs bucket1key2 = createKey(VOLUME_NAME, BUCKET_NAME_ONE,
+        "bucket1key2");
+
+    // Key 1 cannot be reclaimed as it is still referenced by Snapshot 1.
+    writeClient.deleteKey(bucket1key1);
+    // Key 2 is deleted here, which means we can reclaim
+    // it when snapshot 2 is deleted.
+    writeClient.deleteKey(bucket1key2);
+
+    createSnapshot(VOLUME_NAME, BUCKET_NAME_ONE, "bucket1snap2",
+        ++snapshotCount);
+    createKey(VOLUME_NAME, BUCKET_NAME_ONE, "bucket1key4");
+    OmKeyArgs bucket1key5 = createKey(VOLUME_NAME, BUCKET_NAME_ONE,
+        "bucket1key5");
+    writeClient.deleteKey(bucket1key5);
+
+    createSnapshot(VOLUME_NAME, BUCKET_NAME_ONE, "bucket1snap3",
+        ++snapshotCount);
+
+    String snapshotKey2 = "/vol1/bucket1/bucket1snap2";
+    SnapshotInfo snapshotInfo = om.getMetadataManager()
+        .getSnapshotInfoTable().get(snapshotKey2);
+
+    snapshotInfo
+        .setSnapshotStatus(SnapshotInfo.SnapshotStatus.SNAPSHOT_DELETED);
+    om.getMetadataManager()
+        .getSnapshotInfoTable().put(snapshotKey2, snapshotInfo);
+    snapshotInfo = om.getMetadataManager()
+        .getSnapshotInfoTable().get(snapshotKey2);
+    Assertions.assertEquals(snapshotInfo.getSnapshotStatus(),
+        SnapshotInfo.SnapshotStatus.SNAPSHOT_DELETED);
+    return snapshotCount;
+  }
+
+  private OmKeyArgs createKey(String volumeName, String bucketName,
+       String keyName) throws IOException {
+    OmKeyArgs keyArg =
+        new OmKeyArgs.Builder()
+            .setVolumeName(volumeName)
+            .setBucketName(bucketName)
+            .setKeyName(keyName)
+            .setAcls(Collections.emptyList())
+            .setReplicationConfig(StandaloneReplicationConfig.getInstance(
+                HddsProtos.ReplicationFactor.ONE))
+            .setLocationInfoList(new ArrayList<>())
+            .build();
+
+    // Open and write the key.
+    OpenKeySession session = writeClient.openKey(keyArg);
+    writeClient.commitKey(keyArg, session.getId());
+
+    return keyArg;
+  }
+
+  private void createSnapshot(String volName, String bucketName,
+       String snapName, int count) throws Exception {
+    writeClient.createSnapshot(volName, bucketName, snapName);
+
+    GenericTestUtils.waitFor(() -> {
+      try {
+        return omMetadataManager.getSnapshotInfoTable()
+            .getEstimatedKeyCount() >= count;
```

Comment:
Wouldn't it be better to use countRowsInTable(): https://github.com/apache/ozone/blob/7bdfbba56f0f9b47d62b702e32ee846a2701b9fe/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java#L1477

Reply:
Since the test is small it shouldn't be an issue here. From the multiple test runs, I see it is fairly accurate for at least a small number of keys. But yes we can change this. I'll update the PR. Thanks for pointing.